corehq/apps/hqwebapp/models.py

```diff
@@ -141,11 +141,10 @@ class ServerLocation:
         },
     }
 
-    ENVS = DATA.keys()
-    SUBDOMAINS = {env: server['subdomain'] for env, server in DATA.items()}
+    SUBDOMAINS = {env: server['subdomain'] for env, server in ENVS.items()}
 
     @classmethod
     def sorted_form_choices(cls):
-        env = settings.SERVER_ENVIRONMENT
-        sorted_keys = sorted(cls.ENVS, key=lambda x: x != env)
-        return [(cls.DATA[key]['subdomain'], cls.DATA[key]['long_name']) for key in sorted_keys]
+        current_env = settings.SERVER_ENVIRONMENT
+        sorted_keys = sorted(cls.ENVS.keys(), key=lambda x: x != current_env)
```

Comment:
Just realized -- is this intentional? the lambda here reads like its intended to be a filter, but `sorted` will not treat it that way. It will just turn every element into a boolean and order everything that isn't the current environment above or beneath the current environment, in a random order.

Reply:
The intent here is to sort the current environment to the top of choices, and everything else after it in its original order (which will be stored in ServerLocation alphabetically), e.g.:
```
>>> data = ('a', 'b', 'c', 'd')
>>> sorted(data, key=lambda x: x != 'c')
['c', 'a', 'b', 'd']
```

Reply:
If it's any clearer, I could also do this like:
```python
current_env = settings.SERVER_ENVIRONMENT
sorted_envs = sorted(cls.ENVS.items(), key=lambda x: x[0] != current_env)
return [(env_data['subdomain'], env_data['long_name']) for __, env_data in sorted_envs]
```

Reply:
I also had to think for a second what this is doing. It makes sense though, and I think it's easier to understand than the other more verbose options.

Reply:
Eh, I tend to prefer the more explicit last approach. Technically, yes, this does work and it complies with the `sorted` API, but it felt like a pretty unintuitive use for `sorted`. If the task, in English, is to move the current environment to the top of the list, I'd prefer to see code that reflects that. That said, if this is just going to be rewritten anyway, I am not concerned about it